src/main/java/com/rackspacecloud/metrics/queryservice/services/QueryServiceImpl.java

```diff
@@ -140,11 +142,17 @@ private QueryResult getDataPointsForGivenQuery(String tenantId, String queryStri
             );
         }
 
-        String[] strArr = trimmedString.split("from|FROM");
-        if(strArr.length < 2) return null;
+        String regexString = "from|FROM";
```

Comment:
You should be able to enable case-insensitive matching in regexes using the (?i) or (?ui) flag in the beginning of the regex as per https://docs.oracle.com/javase/1.5.0/docs/api/java/util/regex/Pattern.html

This would be to avoid "From" failing the regex

Reply:
👍